runtime/expr.h

```diff
@@ -457,8 +457,12 @@ class eval_context;
 // Call `target`.
 class call_stmt : public stmt_node<call_stmt> {
 public:
-  typedef index_t (*callable_t)(eval_context&);
-  using callable = std::function<index_t(eval_context&)>;
+  struct callable {
+    std::function<index_t(eval_context&)> fn;
```

Comment:
If the return value is an error code, we might do well do go ahead and defined it as a bespoke `error_t` enum or similar rather than just a naked int-ish type.

Reply:
The design currently is that this shares a type with the result of any node in the IR, including exprs, which is index_t.